plugins/modules/gitlab_merge_request.py

```diff
@@ -0,0 +1,415 @@
+            ['state', 'absent', ['source_branch', 'target_branch', 'title'], True],
+        ],
+        supports_check_mode=True
+    )
+    ensure_gitlab_package(module)
+
+    project = module.params['project']
+    source_branch = module.params['source_branch']
+    target_branch = module.params['target_branch']
+    title = module.params['title']
+    description = module.params['description']
+    labels = module.params['labels']
+    description_path = module.params['description_path']
+    remove_source_branch = module.params['remove_source_branch']
+    state_filter = module.params['state_filter']
+    assignee_ids = module.params['assignee_ids']
+    reviewer_ids = module.params['reviewer_ids']
+    state = module.params['state']
+
+    gitlab_version = gitlab.__version__
+    if LooseVersion(gitlab_version) < LooseVersion('2.3.0'):
+        module.fail_json(msg="community.general.gitlab_merge_request requires python-gitlab Python module >= 2.3.0 (installed version: [%s])."
+                             " Please upgrade python-gitlab to version 2.3.0 or above." % gitlab_version)
+
+    gitlab_instance = gitlab_authentication(module)
+    this_gitlab = GitlabMergeRequest(module=module, project=project, gitlab_instance=gitlab_instance)
+
+    r_source_branch = this_gitlab.get_branch(source_branch)
+    if not r_source_branch:
+        module.fail_json(msg="Source branch {b} not exist.".format(b=r_source_branch))
+
+    r_target_branch = this_gitlab.get_branch(target_branch)
+    if not r_target_branch:
+        module.fail_json(msg="Destination branch {b} not exist.".format(b=r_target_branch))
+
+    this_mr = this_gitlab.get_mr(title, source_branch, target_branch, state_filter)
+
+    if state == "present":
+        if description_path:
+            try:
+                f = open(description_path, 'rb')
+                description = to_text(f.read(), errors='surrogate_or_strict')
+                f.close()
+            except IOError as e:
+                module.fail_json(msg='Cannot open {0}: {1}'.format(description_path, e))
+
+        assignee_ids = sorted(this_gitlab.get_user_ids(assignee_ids.split(","))) if assignee_ids else []
+        reviewer_ids = sorted(this_gitlab.get_user_ids(reviewer_ids.split(","))) if reviewer_ids else []
+        labels = sorted(labels.split(",")) if labels else []
+
+        options = {
+            "target_branch": target_branch,
+            "title": title,
+            "description": description,
+            "labels": labels,
+            "remove_source_branch": remove_source_branch,
+            "reviewer_ids": reviewer_ids,
+            "assignee_ids": assignee_ids,
+        }
+
+        if not this_mr:
+            options["source_branch"] = source_branch
+
+            mr = this_gitlab.create_mr(options)
+            module.exit_json(
+                changed=True, msg="Created the Merge Request {t} from branch {d} to branch {s}.".format(t=title, d=target_branch, s=source_branch),
```

Comment:
Source and target branches seem to be inverted.
```suggestion
                changed=True, msg="Created the Merge Request {t} from branch {s} to branch {d}.".format(t=title, d=target_branch, s=source_branch),
```